### DISCUSSION TITLE
Bitwarden Passwords via CLI

### DISCUSSION THREAD

**Discussion by ScytheDraven47** [johnlindquist/kit#1251] [Share]
I figured it'd be nice to have a Bitwarden Script for use outside of browsers, and it made for a good first mini project.
It uses [@bitwarden/cli](https://www.npmjs.com/package/@bitwarden/cli) via NPM, though I'm looking at doing an API version as well.

[Code/gist here](https://gist.github.com/ScytheDraven47/0605ea9475778ae9cc2279c6fd07ad2e)

`Enter` copies password
`Ctrl+Enter` copies username
`Ctrl+Shift+Enter` pastes username, then tabs once, then pastes password (won't work for all use cases, but figured it's nice to have)

This script does not save user credentials, but saves a session key to prevent frequent logging in.

Currently missing 2FA via email and YubiKey.

**Comment by timlindnet** [comment]
Using the script with Scriptkit (Windows 10) version 1.99.82 I'm getting:
`Command failed with exit code 1: bw status
'bw' is not recognized as an internal or external command,
operable program or batch file.
Error: Command failed with exit code 1: bw status
'bw' is not recognized as an internal or external command,
operable program or batch file.
    at makeError (file:///C:/Users/litmind/.kit/node_modules/@johnlindquist/globals/dist/index.js:36:111645)
    at file:///C:/Users/litmind/.kit/node_modules/@johnlindquist/globals/dist/index.js:36:127769
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)`
    
LOGGING: true --> 
checking status...
uninstalled
requesting install...
